packages/admin-panel/src/layout/navigation/SecondaryNavbar.jsx

```diff
@@ -0,0 +1,262 @@
+    const rightOverflow = !getIsElementVisible(
+      navLinkRefs.current[navLinkRefs.current.length - 1].current,
+    );
+
+    setOverflows({
+      left: leftOverflow,
+      right: rightOverflow,
+    });
+  };
+
+  const getIsElementVisible = (element, buffer = 20) => {
+    if (!element) return false;
+
+    const boundingRect = element.getBoundingClientRect();
+    const containerBoundingRect = containerRef.current.getBoundingClientRect();
+    const { left, right } = boundingRect;
+    const { left: containerLeft, right: containerRight } = containerBoundingRect;
+
+    // include a buffer to account say that within 20px of the edge is still considered visible
+    const isVisible = left >= containerLeft - buffer && right <= containerRight + buffer;
+    return isVisible;
+  };
+
+  const handleScroll = item => {
+    if (!item) return;
+
+    // scroll to the item, with smooth behavior and aligning to the start of the container
+    item.scrollIntoView({ behavior: 'smooth', block: 'nearest', inline: 'start' });
+  };
+
+  const scrollToNextVisibleItem = () => {
+    // get the currently visible elements
+    const visibleElements = navLinkRefs.current.filter(ref => getIsElementVisible(ref.current));
+
+    // get the index of the last visible element
+    const lastVisibleElementIndex = navLinkRefs.current.indexOf(
+      visibleElements[visibleElements.length - 1],
+    );
+
+    // the next item to scroll to is the one after the last visible element
+    const nextIndex = lastVisibleElementIndex + 1;
+    const nextItem = navLinkRefs.current[nextIndex];
+    if (!nextItem) return;
+    handleScroll(nextItem.current);
+  };
+
+  const scrollToPrevVisibleItem = () => {
+    // get the currently visible elements
+    const visibleElements = navLinkRefs.current.filter(ref => getIsElementVisible(ref.current, 50));
+    // get the index of the first visible element
+    const firstVisibleElementIndex = navLinkRefs.current.indexOf(visibleElements[0]);
+    // the previous item to scroll to is the current first visible element minus the number of elements to scroll or 0 if it's negative
+    const prevIndex = Math.max(firstVisibleElementIndex - numToScroll, 0);
+
+    const prevItem = navLinkRefs.current[prevIndex];
+    if (!prevItem) return;
+    handleScroll(prevItem.current);
+  };
+
+  return {
+    scrollToNextVisibleItem,
+    scrollToPrevVisibleItem,
+    overflows,
+  };
+};
+
+export const SecondaryNavbar = ({ links: linkInput, baseRoute }) => {
+  const containerRef = useRef(null);
+  const navLinkRefs = useRef(linkInput.map(() => React.createRef()));
+  const links = linkInput.map(link => ({
+    ...link,
+    target: link.exact ? link.to : `${baseRoute}${link.to}`,
+  }));
+
+  const { scrollToNextVisibleItem, scrollToPrevVisibleItem, overflows } = useScrollableMenu(
+    containerRef,
+    navLinkRefs,
+  );
+
+  return (
+    <Wrapper>
+      {overflows?.left && (
```

Comment:
This `overflows` uses optional chaining, but line 245 doesn’t. Should they be consistent with each other?